components/icons/test/icon.vdiff.js

```diff
@@ -0,0 +1,51 @@
+import '../../colors/colors.js';
+import '../icon.js';
+import { expect, fixture, html } from '@brightspace-ui/testing';
+
+describe('d2l-icon', () => {
+
+	['tier1', 'tier2', 'tier3'].forEach((tier) => {
+		const template = html`<d2l-icon icon="${tier}:assignments"></d2l-icon>`;
+		it(tier, async() => {
+			const elem = await fixture(template);
+			await expect(elem).to.be.golden();
+		});
+		it(`rtl-${tier}`, async() => {
+			const elem = await fixture(template, { rtl: true });
+			await expect(elem).to.be.golden();
+		});
+	});
+
+	[
+		{
+			name: 'prefixed',
+			template: html`<d2l-icon icon="d2l-tier3:assignments"></d2l-icon>`
+		},
+		{
+			name: 'fill-none',
+			template: html`<d2l-icon icon="tier2:evaluate-all"></d2l-icon>`
+		},
+		{
+			name: 'fill-circle',
+			template: html`<d2l-icon icon="tier2:divider-big"></d2l-icon>`
+		},
```

Comment:
I chose not to use the demo color/size components in here since they seemed to be obfuscating more than they were contributing.